platform/darwin/src/MGLVectorStyleLayer.h

```diff
@@ -128,6 +128,22 @@ NS_ASSUME_NONNULL_BEGIN
  style attributes and also `hyphen-minus` and `tag:subtag`. However, you must use
  `%K` in the format string to represent these variables:
  `@"%K == 'LineString'", @"$type"`.
+ 
+ ### Example ###
+ To create a filter with the logic `(index == 10 || index == 5) && ele >= 200`, 
+ you could create a predicate using `NSCompoundPredicate` along these lines:
+ 
+ ```swift
+ let meters = 1500.0
+ let layer = MGLLineStyleLayer(identifier: "contour", source: contours)
+ layer.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [
```

Comment:
This is a lot of typing. Is there a point you want to make about the individual classes within the NSPredicate class cluster, or would a single predicate format string such as `(index == 10 || index == 5) && ele >= 200` be sufficient?